File: app/src/main/java/com/leadinsource/dailyweightlog/db/WeightDao.java
package com.leadinsource.dailyweightlog.db;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Query;
import android.database.Cursor;

import java.util.List;

/**
 * Created by Matt on 23/02/2018.
 */

@Dao
public interface WeightDao {
    @Query("SELECT * FROM "+ DataContract.WeightEntry.TABLE_NAME + " ORDER BY "+ DataContract.WeightEntry.COLUMN_DATE)
    Cursor getAll();

    @Query("SELECT * FROM "+DataContract.WeightEntry.TABLE_NAME + " WHERE "+ DataContract.WeightEntry._ID +" = :weightId")
    LiveData<Weight> getSingleWeight(int weightId);

}

File: app/src/main/java/com/leadinsource/dailyweightlog/db/WeightDao.kt
package com.leadinsource.dailyweightlog.db

import androidx.room.Dao
import androidx.room.Query
import androidx.room.Insert

/**
 * Created by Matt on 23/02/2018.
 */

@Dao
interface WeightDao {

    @Query("SELECT * FROM weight")
    suspend fun getAll(): List<Weight>

    @Insert
    suspend fun insert(weight: Weight)
/*
    @Query("SELECT * FROM $TABLE_NAME ORDER BY $COLUMN_DATE")
    fun getAll(): List<Weight>

    @Query("SELECT * FROM $TABLE_NAME WHERE $_ID = :weightId")
    fun getSingleWeight(weightId: Int): List<Weight>*/
}
